### PR TITLE
Issue 3859: Fix scale_segmentStore script for external access

### LINKED ISSUE #3859: IO is not getting started with 3 SS and 4 Node Swarm setup
**After setting up Docker Swarm setup with 4 nodes(1 Master & 3 Workers), IO is not getting started.**

All 3 Segments stores are getting distributed in all 3 nodes(out of 4) but as per segment store logs, we observed that all the 4 containers 0-3 are getting assigned to all 3 segment store. 
```
2019-05-30 10:29:43,846 401529 [core-8] INFO  i.p.s.s.h.ZKSegmentContainerMonitor - Container Changes: Desired = [0, 1, 2, 3], Current = [0, 1, 2, 3], PendingTasks = [], ToStart = [], ToStop = [].
```
It looks like controller is not properly distributing all 4 containers to 3 segment stores and hence segment store is not functioning properly.

Following is the Swarm setup deployed as per the document procedure:
```
3 Segment Store
2 Controller
4 Bookkeeper
1 Zookeeper
```

Attaching the logs for all the containers.
[4node3ssLog.zip](https://github.com/pravega/pravega/files/3239921/4node3ssLog.zip)

### PR DESCRIPTION
**Pull request by shrids**
**Change log description**  
* Ensures `publishedIPAddress` configuration is read from the docker configuration.
* Adds missing configuration for HDFS.

**Purpose of the change**  
Fixes #3859 #3874 

**What the code does**  
`scale_segmentStore ` is a helper script which is used to scale segment stores (with HDFS as tier2) instances with external access. This enables Pravega to be accessed by clients outside the docker swarm cluster. The user uses the system property `PUBLISHED_ADDRESS` to convey the ip to be used by the external clients to access Pravega.

The script ensures it reads the configuration from the running segment store docker service 
via `docker service inspect pravega_segmentstore` and uses the same properties to create newer segment store service instances.

**How to verify it**  
Verified it by testing it on a docker swarm setup.

### CODE REVIEW

**Comment by shrids** [comment]
Verified it with pravega version `0.5.0-2260.b477eec`

File: docker/compose/swarm/scale_segmentstore [hunk 1]
```diff
@@ -17,7 +17,7 @@ try:
         "docker service inspect pravega_segmentstore --format='{{.Spec.TaskTemplate.ContainerSpec.Env}}'", shell=True)
 
     def get_config(name):
-        return re.findall(name + "=(.+?)( |])", env)[0][0]
+        return re.findall(name + "=(.+?)( |]|\r\n|\n)", env)[0][0]
```
Comment (fpj) [inline docker/compose/swarm/scale_segmentstore]: One clarification question about the second part of the regular expression: `( |]|\r\n|\n)`. The first part is making sure that after the `=` sign there is a string with one or more `.`.  The second part is a disjunction with four options:

1- Space
2- Closing bracket `]`
3- End of line `\r\n`
4- Line feed `\n`

I suppose that (1) is for the case the user accidentally leaves a space after the address. I'm not sure about (2), though. (3) and (4) make sense.

Could you clarify (1) and (2)?

Comment (shrids) [reply docker/compose/swarm/scale_segmentstore]: The output [1] of `docker service inspect` command executed by the `scale_segmentstore` can include `]`   e.g: `ZK_URL=zookeeper:2181]` . 
An example of the space included is `HDFS_URL=hdfs://hdfs:8020 `.

```
docker service inspect pravega_segmentstore --format='{{.Spec.TaskTemplate.ContainerSpec.Env}}'
[CONTROLLER_URL=tcp://controller:9090 HDFS_REPLICATION=1 HDFS_URL=hdfs://hdfs:8020 JAVA_OPTS=-Dmetrics.enableCSVReporter=false
-Dpravegaservice.publishedIPAddress=10.0.2.15
-Dpravegaservice.listeningIPAddress=0
-Dbookkeeper.bkEnsembleSize=2
-Dbookkeeper.bkAckQuorumSize=2
-Dbookkeeper.bkWriteQuorumSize=2
-Xmx900m
-XX:OnError="kill -9 p%"
-XX:+ExitOnOutOfMemoryError
-XX:+CrashOnOutOfMemoryError
-XX:+HeapDumpOnOutOfMemoryError
 TIER2_STORAGE=HDFS WAIT_FOR=bookkeeper:3181,hdfs:8020 ZK_URL=zookeeper:2181]
```

**Comment by Richa-Srivastava** [comment]
Verified in the private build (https://raw.githubusercontent.com/pravega/pravega/0c78ad04e8507375eb1fdfcfb2c7aa10e0571775/docker/compose/swarm/scale_segmentstore) `scale_segmentStore` script is working fine with scaling segment stores and distributing segment containers properly to each SS. Also, no HDFS writeup issues found in scaled up SS.

**Comment by sachin-j-joshi** [comment]
LGTM

File: docker/compose/swarm/scale_segmentstore [hunk 2]
```diff
@@ -59,6 +59,8 @@ for instance in to_create:
           --label com.docker.stack.namespace='pravega' \
           -p '{port}:{port}' \
           -e HDFS_URL={hdfs_url} \
+          -e TIER2_STORAGE=HDFS \
```
Comment (fpj) [inline docker/compose/swarm/scale_segmentstore]: Are we hardcoding HDFS as the Tier 2 option we are using for swarm deployments?

Comment (shrids) [reply docker/compose/swarm/scale_segmentstore]: yes, HDFS is the only option supported tier2 storage for docker swarm deployments.

**Pull request merged**